src/graphql/node/control.rs

```diff
@@ -177,13 +177,15 @@ fn node_apply_scope(node: &NodeInput) -> NodeApplyScope {
     let target_agents = if is_any_agent_changed {
         let mut updates = Vec::new();
         let mut disables = Vec::new();
-        node.agents.iter().for_each(|agent| {
-            if agent.draft.is_none() {
-                disables.push(agent.key.as_str());
-            } else if agent.draft != agent.config {
-                updates.push(agent.key.as_str());
-            }
-        });
+        node.agents
+            .iter()
+            .for_each(|agent| match (&agent.draft, &agent.config) {
+                (None, _) => disables.push(agent.key.as_str()),
+                (Some(draft), _) if Some(draft) != agent.config.as_ref() && !draft.is_empty() => {
+                    updates.push(agent.key.as_str());
+                }
+                _ => {}
+            });
```

Comment:
I think it would be simpler to use `fold`.

Reply:
Done